Remove `coeffs` and `ops` from hyperparameters of `PrepSelPrep` and dequeue its `lcu` input (#8169)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Breaking changes 💔</h3>

> * :class:`~.PrepSelPrep` now de-queues its input ``lcu``, making it consistent with other
>   PennyLane operators. In addition, the attributes ``coeffs`` and ``ops`` are now derived from
>   the property ``lcu``, instead of storing them independently. This avoids redundant
>   information storage, which is less memory-efficient and can lead to multiple sources of truth.

Reviewer: ```suggestion
* :class:`~.PrepSelPrep` is now more memory-efficient. This was accomplished by de-queuing its input ``lcu`` to make it consistent with other
  PennyLane operators. In addition, the attributes ``coeffs`` and ``ops`` are now derived from
  the property ``lcu`` instead of storing them independently. 
```
